src/analysis/souffle/operations.dl

```diff
@@ -0,0 +1,80 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+#ifndef SRC_ANALYSIS_SOUFFLE_OPERATIONS_DL_
+#define SRC_ANALYSIS_SOUFFLE_OPERATIONS_DL_
+
+#include "dataflow_graph.dl"
+
+// The symbol representing the operator. For operations commonly represented by
+// a single character operator (such as +, =, /, *, etc), the operator will be
+// the tsring containing that character.
+.type Operator <: symbol
+
+// A list of operands, used for gathering all operators in a single relation
+// regardless of arity. Useful for rules that apply to any operation.
+.type OperandList = [ operand: AccessPath, next: OperandList ]
+
+// Operations of any arity go in this relation.
+.decl operation(owner: Principal, operator: Operator, result: AccessPath, operandList: OperandList)
```

Comment:
My thinking was that if an `operation` is placed into the dataflow graph, it must have been the result of someone claiming that it should be there. This `owner` is that someone, in authorization logic `Principal` form. I think this would be especially useful if we decided that we used these `operation`s to show some partial internal dataflow of the `Particle`, as they would then be claims about the `Particle`'s behavior akin to `derives from`.

Reply:
As discussed on the phone, going with this for now, as it'd be easier to remove it later than add it back in later.